src/components/partner/partner.service.ts

```diff
@@ -225,10 +242,27 @@ export class PartnerService {
         ...secured.pointOfContact,
         value: result.pointOfContactId,
       },
+      types: {
+        ...secured.types,
+        value: secured.types.value || [],
+      },
     };
   }
 
   async update(input: UpdatePartner, session: ISession): Promise<Partner> {
+    const object = await this.readOne(input.id, session);
+    const changes = {
+      ...input,
+      types: uniq(input.types),
```

Comment:
I wasn't going to make a big deal of it before, but I think this would be better as a transform on the DTO.
```tsx
@Transform(uniq)
```